### PR TITLE
feat: read from `Store` db when running evm

### LINKED ISSUE #190: Use `Store` as db for evm execution

### PR DESCRIPTION
**Pull request by fmoletta**
**Motivation**

Being able to read from `Store` database when running the evm

**Description**

Main changes:
* Implements the necessary trait for the evm to be able to read from `Store` when executing
* Add struct `EvmState` that can be built from `Store` and encapsulates the evm-dependant state implementation to be able to switch evm implementation without modifying other crates.

Other:
* Move evm into its own trait to avid cyclic dependencies when importing from storage crate
* Add method `add_account` to `Store` in order to write all account data at once

Closes #190

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ members = [
     "crates/net",
     "crates/rpc",
     "crates/storage",
+    "crates/evm",
```
Comment (mpaulucci) [inline Cargo.toml:8]: why did you extract this into a different crate?

File: cmd/ef_tests/test_runner.rs [hunk 2]
```diff
@@ -33,7 +28,7 @@ fn execute_test(test: &TestUnit) {
             .clone()
             .unwrap()
             .into(),
-        &pre,
+        &mut build_evm_state_from_prestate(&test.pre),
```
Comment (mpaulucci) [inline cmd/ef_tests/test_runner.rs:31]: this will allow us to execute multiple transactions and blocks, right?

**Review by mpaulucci — APPROVED** [review]
great work! 🚀

File: cmd/ef_tests/test_runner.rs (hunk 2 above)
Comment (fmoletta) [reply cmd/ef_tests/test_runner.rs:31]: Yes

**Pull request merged**